### ISSUE TITLE
Document -webkit-text-security

### ISSUE THREAD

**Issue by jfkthame** [whatwg/compat#240]
The non-standard `-webkit-text-security` property exists in both webkit and blink, but does not seem to be mentioned in the compatibility standard.

We've seen real-world webcompat issues because this property is not implemented in Firefox; see https://bugzilla.mozilla.org/show_bug.cgi?id=1826629 and https://github.com/webcompat/web-bugs/issues/114591.

(There's some information about `-webkit-text-security` [on MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-text-security), but note that it is not entirely complete or accurate at present.)

Note also that `-webkit-text-security` is entirely distinct from the ([questionable](https://github.com/w3c/csswg-drafts/issues/6788)) CSS-UI-4 `input-security` property.

**Comment by miketaylr** [comment]
Ah, interesting. If Mozilla is willing to land this, we should totally spec it here. Thanks for the report @jfkthame.

**Comment by jfkthame** [comment]
I don't think Mozilla is exactly "keen" on this -- but we felt it unlikely that webkit and blink would be prepared to drop it, given existing usage. If you think that's a possibility, though, that'd be great.

**Comment by miketaylr** [comment]
Yeah, sadly usage is much higher than the typical threshold for a deprecation: https://chromestatus.com/metrics/css/timeline/popularity/322
![Screenshot 2023-04-14 at 11 47 26 AM](https://user-images.githubusercontent.com/67283/232092305-8b24dc2b-c6e1-4c8f-94a0-97fc239ed572.png)

**Comment by jfkthame** [comment]
Do you know whether that reflects only "direct" usage by authors, or does it also include the use of `<input type=password>`, which at least in blink appears to set `-webkit-text-security` via the UA stylesheet?

If the latter gets counted as "use of this feature", that could substantially change the picture.

**Comment by miketaylr** [comment]
I suspect the former, given that `<input type=password>` usage is much higher (4.5%) https://chromestatus.com/metrics/feature/timeline/popularity/192

**Comment by karlcow** [comment]
https://github.com/search?q=%22-webkit-text-security%22&type=code

it seems also used for fingerprinting, I wonder how it might break firefox in another way, by selecting codepath which was not meant to be seen. 

https://github.com/Roll20/roll20-character-sheets/blob/ba70f6e672622420f899532716e512bfae6a550c/TheWitcher/witchersheet.css#L4710-L4717

but most of the time it seems to be used by itself.

There are some interesting usage patterns in terms of UI on why people are using it or more exactly why in some circumstances people do not want to use `type="password"`
https://stackoverflow.com/questions/25822682/webkit-text-security-compatibility#answer-71149466

1. want to have a hidden PIN code, but do not want the browser to prompt for saving it.
2. The fact to be able to get a UI widget to enter numbers.
3. Usage of specific fonts to hide the code. 

Some hacks people rely on to try to get what they want
https://stackoverflow.com/questions/36935576/how-to-make-input-type-tel-work-as-type-password/36936452#36936452

https://css-tricks.com/the-options-for-password-revealing-inputs/


Also the intent to ship from Mozilla for [css-ui-4] Implement "input-security" CSS property
https://groups.google.com/a/mozilla.org/g/dev-platform/c/vikcWAyyAnc

BUT 

https://drafts.csswg.org/css-ui-4/#input-security

>  The CSS-WG has agreed that while be believe that providing this piece of functionality to users is important, doing it via CSS+JS is the wrong approach, and that instead it should be built into user agents: this needs to work consistently from site to site for it to be discoverable and understandable by users, this needs to work even when JS is turned off, this needs to have consistenly solid accessibility… We therefore intend to remove this from the specification and instead, we would like to see this behavior specified in the HTML specification as part of the interaction model of password fields. Holding off deleting until the situation with HTML is clarified. See https://github.com/w3c/csswg-drafts/issues/6788 and https://github.com/whatwg/html/issues/7293.

* [[css-ui] 'input-security' considered harmful](https://github.com/w3c/csswg-drafts/issues/6788)
* [<input type=password> should provide UI to show/hide its value](https://github.com/whatwg/html/issues/7293)

**Comment by karlcow** [comment]
Mozilla just resolved it as fixed.
https://bugzilla.mozilla.org/show_bug.cgi?id=1826629

**Comment by miketaylr** [comment]
Cool, we should document it then. Patches welcome, etc.:)
